packages/banner/src/Banner.tsx

```diff
@@ -0,0 +1,234 @@
+import React from 'react';
+import ImportantWithCircleIcon from '@leafygreen-ui/icon/dist/ImportantWithCircle';
+import InfoWithCircleIcon from '@leafygreen-ui/icon/dist/InfoWithCircle';
+import WarningIcon from '@leafygreen-ui/icon/dist/Warning';
+import CheckmarkWithCircleIcon from '@leafygreen-ui/icon/dist/CheckmarkWithCircle';
+import XIcon from '@leafygreen-ui/icon/dist/X';
+import { css, cx } from '@leafygreen-ui/emotion';
+import { uiColors } from '@leafygreen-ui/palette';
+
+const Variant = {
+  Info: 'info',
+  Warning: 'warning',
+  Danger: 'danger',
+  Success: 'success',
+} as const;
+
+type Variant = typeof Variant[keyof typeof Variant];
+
+export { Variant };
+
+const baseBannerStyles = css`
+  position: relative;
+  display: flex;
+  min-height: 40px;
+  padding: 9px 12px 9px 20px;
+  border-width: 1px;
+  border-style: solid;
+  border-radius: 6px;
+  font-size: 14px;
+  line-height: 20px;
+
+  &:before {
+    content: '';
+    position: absolute;
+    width: 4px;
+    top: 0;
+    bottom: 0;
+    left: 0px;
+    border-radius: 6px 0px 0px 6px;
+  }
+`;
+
+const leftIconStyles = css`
+  flex-shrink: 0;
+  position: absolute;
```

Comment:
Is there a reason we're doing this with absolute positioning? Seems more brittle than using a flex layout IMO

Reply:
Additionally, the banner already appears to be a flex container that we're not currently using.

Reply:
👍 Will you take a look stylistically that all still looks right when i remove this? I initially did it because I was pixel pushing (the icon feels like 1px too high to me) but if it looks good to you then I'm happy to remove this position setting.